src/main/kotlin/hr/foi/pop/backend/filters/AuthTokenFilter.kt

```diff
@@ -0,0 +1,112 @@
+package hr.foi.pop.backend.filters
+
+import hr.foi.pop.backend.definitions.ApplicationErrorType
+import hr.foi.pop.backend.exceptions.BadJwtFormatException
+import hr.foi.pop.backend.responses.ErrorResponse
+import hr.foi.pop.backend.responses.ResponseSender
+import hr.foi.pop.backend.security.jwt.JwtUtils
+import hr.foi.pop.backend.services.AuthenticationService
+import io.jsonwebtoken.ExpiredJwtException
+import io.jsonwebtoken.MalformedJwtException
+import jakarta.servlet.FilterChain
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher
+import org.springframework.util.StringUtils
+import org.springframework.web.filter.OncePerRequestFilter
+
+
+open class AuthTokenFilter(
+    private val excludedRoutes: List<AntPathRequestMatcher>
+) : OncePerRequestFilter() {
+
+    @Autowired
+    private lateinit var jwtUtils: JwtUtils
+
+    @Autowired
+    private lateinit var authenticationService: AuthenticationService
+
+    override fun shouldNotFilter(request: HttpServletRequest): Boolean {
```

Comment:
I think I found a problem with this part of code, but It is for H2, so myb you want to ignore it, because It will not affect production environment.

Description:
I started the application and went to access the h2-console. Even though you ignored that page in the WebConfigurer class, the filter still activates and throws you 3 JwtFormatException errors. In order to avoid this, you have to add pages that will be ignored within the custom filter. Something similar to this code:

        val excludeUrlForDemonstrationPurpose: ArrayList<String> = ArrayList()
        excludeUrlForDemonstrationPurpose.add("/h2-console/**")

        val pathMatcher: PathMatcher = AntPathMatcher()

        return excludeUrlForDemonstrationPurpose.stream()
            .anyMatch { p ->
                pathMatcher.match(p, request.servletPath)
            }
       
I recommend separating the paths into two static lists so that you can use them both within the filter and within the WebSecurityConfigurer, where one list will have free routes (auth, h2-conosle), and one will have those that need to be authenticated. You can also add special lists depending on the role of the user. It will be easier for you to add them later inside the http.authorizeRequests method.

Example of errors while accessing h2-console:
![image](https://github.com/foivz/pop-app-backend/assets/61520141/d75ea863-da7b-46bb-8523-7d6c4d3d8949)

Reply:
I have kinda ignored this, but yes, I am going to attempt to get this fixed using your method.

Reply:
Fixed via [64d4008](https://github.com/foivz/pop-app-backend/pull/22/commits/64d4008f015fa1e3ed3e4f7afb1132a470fc09a4).